test/unit/test_circuit_runner_job.py

```diff
@@ -34,6 +34,7 @@ def __init__(self, data: Any) -> None:
         """
         self._data = data
         api_client = MagicMock()
+        api_client.job_results = MagicMock(return_value=self._data)
```

Comment:
Why is this line added? It seems to have no relation to the PR we want to revert

Reply:
In `test_circuit_metadata_ndarray` we were mocking `_download_external_result` to just return the data given so since that method has been removed, i'm mocking `job_results()`